syntax = "proto3";

package embodied.unity;

enum RequestSourceType {
    ROBOT_TTS_REQUEST = 0;
    REMOTECHAT_TTS_REQUEST = 1;
}

message TTSMark {
    uint32 time = 1;
    uint32 start = 2;
    uint32 end = 3;
    string type = 4;
    string value = 5;
}

message CloudTTSRequest {
    string markup = 1;
    string event_id = 2;
    int32 chunk_num = 3;
    uint64 timestamp = 4;
    string user_id = 5;

    string software_version = 100;
    string module_name = 101;
}

message AudioBuffer {
    bytes buffer = 1;
    int32 channels = 2;
    int32 sample_rate = 3;
}

message CloudTTSResponse {
    RequestSourceType request_source = 1;
    AudioBuffer audio = 2;
    repeated TTSMark marks = 3;
    string event_id = 4;
    int32 chunk_num = 5;
    uint64 timestamp = 6;
    uint64 total_time = 7;
    uint64 synthesis_time = 8;

    string software_version = 100;
    string module_name = 101;
}

message CloudTTSSupplement {
    uint64 timestamp = 1;
    string event_id = 2;
    int32 chunk_num = 3;
    string text = 4;      // the full text used in synthesis (translation if the target language is not english)
    string markup = 5;    // the full markup passed to the synthesis engine (translation if the target language is not english)
    uint64 translation_time = 6;
    uint64 automarkup_time = 7;
    uint64 synthesis_time = 8;
    string tts_engine = 9;

    uint64 total_time = 20;

    string software_version = 100;
    string module_name = 101;
}
